### PR TITLE
.Net: Fix various code issues and improve error handling, update .editorconfig, and update code in various files

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
## Summary
This pull request includes several changes to improve the code quality and error handling of the project, updates to the .editorconfig file, and updates to code in various files. The changes include fixing a RequestFailedException extension method, updating a ChatWithDataResponse class, fixing a unit test for HuggingFaceEmbeddingGeneration, updating a QdrantMemoryStore test, and fixing a RedisMemoryStore test. Additionally, the OobaboogaTextCompletionTests class has been updated to handle concurrent WebSocket connections more effectively. The .editorconfig file has been updated to enable a naming rule violation warning, suppress unnecessary diagnostics, and add a new task to the cleanup task list. The LogLevel variable has been renamed to s_logLevel, and a new HttpOperationException method has been added. In AsyncEnumerable.cs, the ToEnumerable method now avoids problematic synchronous waits. Additionally, the ToAsyncEnumerable method now has naming rule violations and lacks await operators. In FunctionHelpers.cs, the CallViaKernel method has been renamed to CallViaKernelAsync. In several files in the MsGraph plugin, the using statements for SkillDefinition have been moved to the end of the list.

## Changes
- Fixing a RequestFailedException extension method to suppress exceptions and ensure an HttpOperationException is thrown instead
- Updating the ChatWithDataResponse class to remove an unused attribute and suppress a code analysis warning
- Fixing a unit test for HuggingFaceEmbeddingGeneration to use Assert.Single instead of Assert.Equal(1, ...)
- Updating a QdrantMemoryStore test to use Assert.Equal instead of Assert.Equal<string>
- Fixing a RedisMemoryStore test to use Assert.Equal instead of comparing exception messages
- Updating the OobaboogaTextCompletionTests class to handle concurrent WebSocket connections more effectively
- Enabling a naming rule violation warning in the .editorconfig file
- Suppressing unnecessary diagnostics in the .editorconfig file
- Adding a new task to the cleanup task list in the .editorconfig file
- Renaming the LogLevel variable to s_logLevel
- Adding a new HttpOperationException method
- In AsyncEnumerable.cs, the ToEnumerable method now avoids problematic synchronous waits. The ToAsyncEnumerable method now has naming rule violations and lacks await operators.
- In FunctionHelpers.cs, the CallViaKernel method has been renamed to CallViaKernelAsync.
- In several files in the MsGraph plugin, the using statements for SkillDefinition have been moved to the end of the list.

---
*Powered by [Microsoft Semantic Kernel](https://github.com/microsoft/semantic-kernel)*

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public void CanCreateStepwisePlan(bool useChatModel, string prompt, string expec
     [InlineData(true, "What is the tallest mountain on Earth? How tall is it divided by 2", "Everest")]
     [InlineData(false, "What is the weather in Seattle?", "Seattle", 1)]
     [InlineData(true, "What is the weather in Seattle?", "Seattle", 1)]
-    public async void CanExecuteStepwisePlan(bool useChatModel, string prompt, string partialExpectedAnswer, int expectedMinSteps = 1)
+    public async System.Threading.Tasks.Task CanExecuteStepwisePlan(bool useChatModel, string prompt, string partialExpectedAnswer, int expectedMinSteps = 1)
```
Comment (markwallace-microsoft) [inline dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs:74]: Should there be an `Async` suffix?

File: dotnet/src/SemanticKernel.UnitTests/KernelTests.cs [hunk 2]
```diff
@@ -124,7 +124,7 @@ public void ItImportsSkillsNotCaseSensitive()
     [Theory]
     [InlineData(null, "Assistant is a large language model.")]
     [InlineData("My Chat Prompt", "My Chat Prompt")]
-    public void ItUsesChatSystemPromptWhenProvided(string providedSystemChatPrompt, string expectedSystemChatPrompt)
+    public async Task ItUsesChatSystemPromptWhenProvided(string providedSystemChatPrompt, string expectedSystemChatPrompt)
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.UnitTests/KernelTests.cs:127]: Should this have an `Async` suffix?

File: dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs [hunk 3]
```diff
@@ -116,7 +117,181 @@ public void ShouldReplacePathParametersByDefaultValues()
...
+
+        // Act
+        var url = sut.BuildOperationUrl(arguments);
+
+        // Assert
+        Assert.Equal("https://fake-random-test-host/?p1=dv1&p2=dv2", url.OriginalString);
+    }
+
+    [Fact]
+    public void ShouldSkipNotRequiredQueryStringParametersIfTheirValuesMissing()
```
Comment (markwallace-microsoft) [inline dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs:194]: nit: `ValuesAreMissing`

File: dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs [hunk 4]
```diff
@@ -116,7 +117,181 @@ public void ShouldReplacePathParametersByDefaultValues()
...
+
+        // Act
+        var url = sut.BuildOperationUrl(arguments);
+
+        // Assert
+        Assert.Equal("https://fake-random-test-host/?p2=v2", url.OriginalString);
+    }
+
+    [Fact]
+    public void ShouldThrowExceptionIfNoValueIsProvideForRequiredQueryStringParameter()
```
Comment (markwallace-microsoft) [inline dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs:231]: nit: `IsProvidedFor`

**Review by markwallace-microsoft — APPROVED** [review]
Couple of minor comments and questions but other LGTM.

There are some great changes in this PR.

File: dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs (hunk 1 above)
Comment (lemillermicrosoft) [reply dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs:74]: I made it so naming rules weren't enforced in test/sample code. But I'd gladly turn that on if folks are comfortable with that requirement.

File: dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs (hunk 4 above)
Comment (lemillermicrosoft) [reply dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs]: From a bad merge resolution, removed.

File: dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs (hunk 3 above)
Comment (lemillermicrosoft) [reply dotnet/src/Skills/Skills.UnitTests/Connectors/WebApi/Rest/RestApiOperationTests.cs]: From a bad merge resolution, removed.

File: dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs (hunk 1 above)
Comment (lemillermicrosoft) [reply dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs:74]: There are 82 violations of IDE1006 across test and sample code.

File: dotnet/src/Connectors/Connectors.UnitTests/.editorconfig [hunk 5]
```diff
@@ -3,4 +3,4 @@
 dotnet_diagnostic.CA2007.severity = none # Do not directly await a Task
 dotnet_diagnostic.VSTHRD111.severity = none # Use .ConfigureAwait(bool) is hidden by default, set to none to prevent IDE from changing on autosave
 dotnet_diagnostic.CS1591.severity = none # Missing XML comment for publicly visible type or member
-
+dotnet_diagnostic.IDE1006.severity = none # Naming rule violations
```
[the remark this answers is not in the document]
Comment (lemillermicrosoft) [reply dotnet/src/Connectors/Connectors.UnitTests/.editorconfig]: Applied these changes.

**Review by shawncal — APPROVED** [review]

**Review by shawncal — APPROVED** [review]

**Pull request merged**